### PR TITLE
feat: allow disabling of ssl

### PR DESCRIPTION
**Pull request by icco**
Finishes https://github.com/h3poteto/aws-global-accelerator-controller/issues/207

### CODE REVIEW

File: stable/aws-global-accelerator-controller/values.yaml [hunk 1]
```diff
@@ -53,7 +53,14 @@ affinity: {}
 
 webhook:
   enabled: true
-  certificateName: aws-global-accelerator-cert
+  port: 8443
   serviceName: aws-global-accelerator-validation
-  issuerName: issuer
-  certificateSecretName: aws-global-accelerator-tls
+  https:
+    enabled: true
```
Comment (h3poteto) [inline stable/aws-global-accelerator-controller/values.yaml:59]: Why do you prepare two parameters?
- `https.enabled`
- `certManager.enabled`

Are you supposed to use `https` with disabled and `certManager` with enabled or `https` with enabled` and `certManager` with disabled?

**Review by h3poteto — COMMENTED** [review]
Probably, HTTP webhook will not work with this PR. Because WebhookConfiguration sends requests with HTTPS as default.
https://github.com/h3poteto/charts/blob/023ac9505105898d64ff7ac021e50bcdbb083e92/stable/aws-global-accelerator-controller/templates/validatingwebhookconfiguration.yaml#L11-L15

File: stable/aws-global-accelerator-controller/values.yaml (hunk 1 above)
Comment (icco) [reply stable/aws-global-accelerator-controller/values.yaml:59]: You could use https without certmanager, but it wouldn't be possible to use certmanager without https. I could nest certmanager inside of the https option?

**Comment by icco** [comment]
> Probably, HTTP webhook will not work with this PR. Because WebhookConfiguration sends requests with HTTPS as default.
> 
> https://github.com/h3poteto/charts/blob/023ac9505105898d64ff7ac021e50bcdbb083e92/stable/aws-global-accelerator-controller/templates/validatingwebhookconfiguration.yaml#L11-L15

https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/ you're right, whelp I guess no webhooks for me. Sorry for all of the confusion. I'll need to figure out what alb-controller does that it can run without any certificates

**Comment by h3poteto** [comment]
> I'll need to figure out what alb-controller does that it can run without any certificates

alb-controller generates the certificate itself if you don't specify cert-manager option. However, of course, the controller needs to create certificates and manage certificate expiration. So many controllers require cert-manager.

File: stable/aws-global-accelerator-controller/values.yaml (hunk 1 above)
Comment (h3poteto) [reply stable/aws-global-accelerator-controller/values.yaml:59]: Ah, sorry, you're right. If we specify certain certificates, we can use https without cert-manager.

**Comment by icco** [comment]
We use a bunch of controllers and yours is the first that's needed certmanager. TIL that they are all generating their own certs. Thanks for the education.

**Pull request closed without merging**